bin/specextract

```diff
@@ -2370,60 +2264,55 @@ def specextract(args):
     #       If solution, it can have
     #       1 or more elements per source file.
 
-    # ~~~ NEW Nov. 20, 2012 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     if asp and asp not in [""," ","None","none","NONE"]:
-    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
         asp_stk = stk.build(asp)
         check_files(asp_stk,"aspect")
         asp_count = len(asp_stk)
 
-
+    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+    # 
     #    i)  dtffile stack; ensure it has
     #        either 1 or src_count elements
 
-    # ~~~ NEW Sep. 27, 2013 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     if dtffile and dtffile not in [""," ","None","none","NONE"]:
-    # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
         dtf_stk = stk.build(dtffile)
         check_files(dtf_stk,"dead time factor")
         dtf_count = len(dtf_stk)
 
         if 1 != dtf_count and src_count != dtf_count:
-            raise IOError("Error: dtffile stack must have either 1 element or the same number of elements as the source stack.  Source stack={0}     dtffile stack={1}".format(src_count,dtf_count))
-    # ~~~ NEW Sep. 27, 2013 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+            raise IOError(f"Error: dtffile stack must have either 1 element or the same number of elements as the source stack.  Source stack={src_count}     dtffile stack={dtf_count}")
+
     else:
         dtf_stk=[""]
         dtf_count = len(dtf_stk)
+        
     # ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
-
-
+    #
     #
     # check that there are no spaces in ancillary file paths, if specified
     #
     for path in asp_stk:
```

Comment:
this could all be put into a loop as you did earlier, rather than having lots of repeated code.